### PR TITLE
Add an option to enable logging data to a file

### PR DESCRIPTION
**Pull request by washingtoneg**

### CODE REVIEW

File: catchpoint_opentsdb_bridge.py [hunk 1]
```diff
@@ -90,26 +108,31 @@ def post(self):
...
+            self._tsdb.myproto.put('catchpoint.rtt.by_region', D, 'rtt', ['nodeid', 'region'])
+            self._tsdb.myproto.put('catchpoint.status.by_region', D, 'counter', ['nodeid', 'region', 'status'])
+            self._tsdb.myproto.put('catchpoint.rtt.by_isp', D, 'rtt', ['nodeid', 'isp'])
+            self._tsdb.myproto.put('catchpoint.status.by_isp', D, 'counter', ['nodeid', 'isp', 'status'])
+            self._tsdb.myproto.put('catchpoint.error.by_node', D, 'counter', ['nodeid', 'error'])
+
+            defer.succeed(True)
+        else:
+            # not connected to tsdb, dump data to file
+            with open(log_file, 'a') as outfile:
```
Comment (supernomad) [inline catchpoint_opentsdb_bridge.py]: should this be wrapped in a try/except block incase the file can't be opened for whatever reason?

File: catchpoint_opentsdb_bridge.py [hunk 2]
```diff
@@ -119,29 +142,21 @@ def post(self):
...
-    tsdb = OpenTSDBFactory()
-    reactor.connectTCP(args.tsdb_host, args.tsdb_port, tsdb)
+    if args.file_mode:
+        l.info('File mode detected. Not connecting to OpenTSDB')
+        l.info("Printing Catchpoint data to local log file {log_file}".format(log_file=log_file))
+        tsdb = None
+    else:
+        l.info('Connecting to OpenTSDB')
+        tsdb = OpenTSDBFactory()
+        reactor.connectTCP(args.tsdb_host, args.tsdb_port, tsdb)
```
Comment (supernomad) [inline catchpoint_opentsdb_bridge.py:165]: Not sure on the requirements here, but any chance we would want to TEE the data between both tsdb and a file?

**Review by supernomad — CHANGES_REQUESTED** [review]
One thought, and a quick question, but looking dope

File: catchpoint_opentsdb_bridge.py (hunk 1 above)
Comment (washingtoneg) [reply catchpoint_opentsdb_bridge.py]: That would be a reasonable thing to do. I'll add a check to ensure that we catch / deal with the exception that'll get thrown if we can't open a handle on the log file.

File: catchpoint_opentsdb_bridge.py (hunk 2 above)
Comment (washingtoneg) [reply catchpoint_opentsdb_bridge.py:165]: The requirements that I have are to write to a logfile for testing and OpenTSDB maintenance windows (or in the event of replacement of that data store). I can see the need to write to both places, though I don't have a use case for that, but I can add that functionality if need be.

Also, I need to update the README to describe the end state of whatever gets implemented.

Thoughts?

Comment (supernomad) [reply catchpoint_opentsdb_bridge.py:165]: ah ok no worries at all, I was just curious what the actual end goal of it was, if there is no need to do the TEE then its probably worth it.

File: catchpoint_opentsdb_bridge.py (hunk 1 above)
Comment (washingtoneg) [reply catchpoint_opentsdb_bridge.py]: I added b21ab79 to address this.

**Review by supernomad — APPROVED** [review]

**Pull request merged**